### ISSUE TITLE
error message at recompile step

### ISSUE THREAD

**Issue by PaulXiCao** [timholy/Revise.jl#399]
I run the following code which throws an error and stacktrace

```julia
julia> using Revise
julia> using SpecialFunctions
[ Info: Recompiling stale cache file /home/<user>/.julia/compiled/v1.2/SpecialFunctions/78gOt.ji for SpecialFunctions [276daf66-3868-5448-9aa4-cd146d93841b]

#
# change something in the SpecialFunctions module
#

julia> using SpecialFunctions
┌ Error: Failed to revise /home/<user>/.julia/dev/SpecialFunctions/src/10_bessel.jl
│   exception =
│    BoundsError: attempt to access 114-element Array{Array{Int64,1},1} at index [0]
│    Stacktrace:
│     [1] getindex at ./array.jl:728 [inlined]
│     [2] push! at /home/<user>/.julia/packages/Revise/0KQ7U/src/backedges.jl:54 [inlined]
│     [3] add_to_backedges!(::Revise.BackEdges, ::Array{Revise.SlotDep,1}, ::Int64, ::JuliaInterpreter.SlotNumber) at /home/<user>/.julia/packages/Revise/0KQ7U/src/backedges.jl:80
└ @ Revise ~/.julia/packages/Revise/0KQ7U/src/Revise.jl:614
┌ Warning: Due to a previously reported error, the running code does not match saved version for the following files:
│   /home/<user>/.julia/dev/SpecialFunctions/src/10_bessel.jl
└ @ Revise ~/.julia/packages/Revise/0KQ7U/src/Revise.jl:622
```

I run the standard julia package from my linux distro
```julia
julia> versioninfo()
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i3-5005U CPU @ 2.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, broadwell)
```

**Comment by goggle** [comment]
From that error message, it seems like you caused that error by *change something in the SpecialFunctions module*.

More specifically, you seem to try to index an array at index `0`, but Julia starts counting array indices at `1`.

**Comment by timholy** [comment]
Nah, it was a bug in Revise itself. @PaulXiCao, sorry for the inconvenience. Fixed in #401, which I'll tag as a new release once merged.

**Comment by timholy** [comment]
I'm going to hold off a day or so tagging a new release, see #400.
